pandas/tests/generic/test_frame.py

```diff
@@ -280,3 +281,30 @@ def test_deepcopy_empty(self):
         empty_frame_copy = deepcopy(empty_frame)
 
         self._compare(empty_frame_copy, empty_frame)
+
+    def test_datetime_after_setitem_with_at(self):
+        # This test covers the unexpected behaviour of datetimeField when using
+        # setitem on another column as reported in issue #6942
+
+        start = pd.to_datetime("20140401")
+
+        result = pd.DataFrame(
+            index=pd.date_range(start, periods=1), columns=["timenow", "Live"]
+        )
+
+        result.at[start, "timenow"] = datetime.today()
```

Comment:
is the "at" call being tested this one or the one below?  if not this one, can you set this with a better-tested indexer (or just directly in the constructor call above)

Reply:
@jbrockmendel yeah that could be done. I did that just to replicate the situation in the original issue thread as it mentioned that the "at" worked before setting an entire column(i.e the "Live" column) but not afterwards.